src/Abstractions/NexusMods.Abstractions.Loadouts.Synchronizers/ALoadoutSynchronizer.cs

```diff
@@ -137,50 +137,55 @@ public async Task<DiskStateTree> FileTreeToDisk(FileTree fileTree, Loadout loado
             await foreach (var entry in _hashCache.IndexFolderAsync(location))
             {
                 var gamePath = installation.LocationsRegister.ToGamePath(entry.Path);
-                if (prevState.TryGetValue(gamePath, out var prevEntry))
-                {
-                    // If the file has been modified outside of the app since the last apply, we need to ingest it.
-                    if (prevEntry.Item.Value.Hash != entry.Hash)
-                    {
-                        HandleNeedIngest(entry);
-                        throw new UnreachableException("HandleNeedIngest should have thrown");
-                    }
 
-                    // Does the file exist in the new tree?
-                    if (!fileTree.TryGetValue(gamePath, out var newEntry))
-                    {
-                        // Don't update the results here as we'll delete the file in a bit
-                        toDelete.Add(KeyValuePair.Create(gamePath, entry));
-                        continue;
-                    }
+                // Does the file exist in the previous state?
+                if (!prevState.TryGetValue(gamePath, out var prevEntry))
+                {
+                    // File is new, and not in the previous state, so we need to abort and do an ingest
+                    HandleNeedIngest(entry);
+                    throw new UnreachableException("HandleNeedIngest should have thrown");
+                }
 
-                    resultingItems.Add(newEntry.GamePath(), prevEntry.Item.Value);
-                    switch (newEntry.Item.Value!)
-                    {
-                        case StoredFile fa:
-                            // StoredFile files are special cased so we can batch them up and extract them all at once.
-                            // Don't add toExtract to the results yet as we'll need to get the modified file times
-                            // after we extract them
+                // If the file has been modified outside the app since the last apply, we need to ingest it.
+                if (prevEntry.Item.Value.Hash != entry.Hash)
```

Comment:
Useless comment. If you need to explain the condition in the if-statement, use a variable name instead:

```csharp
var wasModified = prevEntry.Item.Value.Hash != entry.Hash;
if (wasModified)
```